ERROR while compiling memtrace_viewer v0.14.1

Hello. While attempting to install `memtrace_viewer.v0.14.1`, I had the following error message:
```
The following actions will be performed:
  ∗ install memtrace_viewer v0.14.1

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved memtrace_viewer.v0.14.1  (cached)
[ERROR] The compilation of memtrace_viewer.v0.14.1 failed at "dune build --profile release --default-target @install . --root . --cache-transport=direct".

#=== ERROR while compiling memtrace_viewer.v0.14.1 ============================#
# context     2.1.2 | linux/x86_64 |  | https://opam.ocaml.org#f860491a
# path        ~/.opam/for_memtrace/.opam-switch/build/memtrace_viewer.v0.14.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build --profile release --default-target @install . --root . --cache-transport=direct
# exit-code   1
# env-file    ~/.opam/log/memtrace_viewer-11381-1fa2ec.env
# output-file ~/.opam/log/memtrace_viewer-11381-1fa2ec.out
### output ###
# [...]
# File "vendor/conduit-async/async/private_ssl_real.ml", line 129, characters 39-50:
# Alert deprecated: Core.sexp_opaque
# [since 2019-03] use [@sexp.opaque] instead
# File "vendor/conduit-async/async/private_ssl_real.ml", line 150, characters 39-50:
# Alert deprecated: Core.sexp_opaque
# [since 2019-03] use [@sexp.opaque] instead
# File "vendor/conduit-async/async/private_ssl_real.ml", line 151, characters 35-46:
# Alert deprecated: Core.sexp_opaque
# [since 2019-03] use [@sexp.opaque] instead
# File "vendor/conduit-async/async/private_ssl_real.ml", line 242, characters 33-44:
# Alert deprecated: Core.sexp_opaque
# [since 2019-03] use [@sexp.opaque] instead



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build memtrace_viewer v0.14.1
└─
```
Since this installation succeeded at some point before, I guess adding some more constraints fixes the issue. The installed dependencies are as follows:
```
# Packages matching: installed
# Name                  # Installed # Synopsis
angstrom                0.15.0      Parser combinators built for speed and memory-efficiency
astring                 0.8.5       Alternative String module for OCaml
base-bigarray           base
base-bytes              base        Bytes library distributed with the OCaml compiler
base-threads            base
base-unix               base
base64                  3.5.0       Base64 encoding for OCaml
bigarray-compat         1.1.0       Compatibility library to use Stdlib.Bigarray when possible
bigstringaf             0.8.0       Bigstring intrinsics and fast blits based on memcpy/memmove
biniou                  1.2.1       Binary data format designed for speed, safety, ease of use and backward compatibility as protocols evolve
cmdliner                1.0.4       Declarative definition of command line interfaces for OCaml
conf-gmp                4           Virtual package relying on a GMP lib system installation
conf-gmp-powm-sec       3           Virtual package relying on a GMP lib with constant-time modular exponentiation
conf-libffi             2.0.0       Virtual package relying on libffi system installation
conf-libssl             3           Virtual package relying on an OpenSSL library system installation
conf-pkg-config         2           Check if pkg-config is installed and create an opam switch local pkgconfig folder
conf-zlib               1           Virtual package relying on zlib
cppo                    1.6.8       Code preprocessor like cpp for OCaml
cryptokit               1.16.1      A library of cryptographic primitives
csexp                   1.5.1       Parsing and printing of S-expressions in Canonical form
cstruct                 6.0.1       Access C-like structures directly from OCaml
ctypes                  0.17.1      Combinators for binding to C libraries without writing any C
ctypes-foreign          0.18.0      Virtual package for enabling the ctypes.foreign subpackage.
domain-name             0.4.0       RFC 1035 Internet domain names
dune                    2.9.3       Fast, portable, and opinionated build system
dune-configurator       2.9.3       Helper library for gathering system configuration
easy-format             1.3.2       High-level and functional interface to the Format module of the OCaml standard library
fmt                     0.9.0       OCaml Format pretty-printer combinators
integers                0.5.1       Various signed and unsigned integer types for OCaml
js_of_ocaml             4.0.0       Compiler from OCaml bytecode to JavaScript
js_of_ocaml-compiler    4.0.0       Compiler from OCaml bytecode to JavaScript
js_of_ocaml-ppx         4.0.0       Compiler from OCaml bytecode to JavaScript
jsonm                   1.0.1       Non-blocking streaming JSON codec for OCaml
lambdasoup              0.7.3       Easy functional HTML scraping and manipulation with CSS selectors
logs                    0.7.0       Logging infrastructure for OCaml
magic-mime              1.2.0       Map filenames to common MIME types
markup                  1.0.2       Error-recovering functional HTML5 and XML parsers and writers
memtrace                0.1.2       Streaming client for Memprof
menhir                  20211230    An LR(1) parser generator
menhirLib               20211230    Runtime support library for parsers generated by Menhir
menhirSdk               20211230    Compile-time library for auxiliary tools related to Menhir
ocaml                   4.11.2      The OCaml compiler (virtual package)
ocaml-base-compiler     4.11.2      Official release 4.11.2
ocaml-compiler-libs     v0.12.4     OCaml compiler libraries repackaged
ocaml-config            1           OCaml Switch Configuration
ocaml-migrate-parsetree 1.8.0       Convert OCaml parsetrees between different versions
ocaml-syntax-shims      1.0.0       Backport new syntax to older OCaml versions
ocamlbuild              0.14.0      OCamlbuild is a build system with builtin rules to easily build most OCaml projects.
ocamlfind               1.9.3       A library manager for OCaml
octavius                1.2.2       Ocamldoc comment syntax parser
ppx_derivers            1.2.1       Shared [@@deriving] plugin registry
ppx_tools_versioned     5.4.0       A variant of ppx_tools based on ocaml-migrate-parsetree
ppxlib                  0.15.0      Standard library for ppx rewriters
re                      1.10.3      RE is a regular expression library for OCaml
result                  1.5         Compatibility Result module
seq                     base        Compatibility package for OCaml's standard iterator type starting from 4.07.
sexplib0                v0.14.0     Library containing the definition of S-expressions and some base converters
spawn                   v0.15.0     Spawning sub-processes
stdlib-shims            0.3.0       Backport some of the new stdlib features to older compiler
stringext               1.6.0       Extra string functions for OCaml
topkg                   1.0.5       The transitory OCaml software packager
tyxml                   4.5.0       A library for building correct HTML and SVG documents
uchar                   0.0.2       Compatibility library for OCaml's Uchar module
uri                     4.2.0       An RFC3986 URI/URL parsing library
uutf                    1.0.2       Non-blocking streaming Unicode codec for OCaml
yojson                  1.7.0       Yojson is an optimized parsing and printing library for the JSON format
zarith                  1.12        Implements arithmetic and logical operations over arbitrary-precision integers
```

--- a/packages/memtrace_viewer/memtrace_viewer.v0.14.1/opam
+++ b/packages/memtrace_viewer/memtrace_viewer.v0.14.1/opam
@@ -24,8 +24,8 @@ depends: [
   "dune" {>= "2.7.1"}
   "dune-configurator" {>= "2.7.1"}
   "fmt" {>= "0.8.8"}
-  "js_of_ocaml" {>= "3.7.0"}
-  "js_of_ocaml-ppx" {>= "3.7.0"}
+  "js_of_ocaml" {>= "3.7.0" & < "4.0.0"}
+  "js_of_ocaml-ppx" {>= "3.7.0" & < "4.0.0"}
   "jsonm" {>= "1.0.1"}
   "lambdasoup" {>= "0.7.1"}
   "logs" {>= "0.7.0"}

